packages/mui-base/src/Slider/Root/useSliderRoot.ts

```diff
@@ -0,0 +1,448 @@
+    y: (event as PointerEvent).clientY,
+  };
+}
+
+export const Identity = (x: any) => x;
+/**
+ *
+ * Demos:
+ *
+ * - [Slider](https://mui.com/base-ui/react-slider/#hooks)
+ *
+ * API:
+ *
+ * - [useSliderRoot API](https://mui.com/base-ui/react-slider/hooks-api/#use-slider-root)
+ */
+function useSliderRoot(parameters: UseSliderParameters): UseSliderReturnValue {
+  const {
+    'aria-labelledby': ariaLabelledby,
+    defaultValue,
+    direction = 'ltr',
+    disabled = false,
+    largeStep = 10,
+    max = 100,
+    min = 0,
+    minStepsBetweenValues = 0,
+    name,
+    onValueChange,
+    onValueCommitted,
+    orientation = 'horizontal',
+    rootRef,
+    scale = Identity,
+    step = 1,
+    tabIndex,
+    value: valueProp,
+  } = parameters;
+
+  // We can't use the :active browser pseudo-classes.
+  // - The active state isn't triggered when clicking on the rail.
+  // - The active state isn't transferred when inversing a range slider.
+  const [active, setActive] = React.useState(-1);
+
+  const [dragging, setDragging] = React.useState(false);
+
+  const controlRef: React.MutableRefObject<HTMLElement | null> = React.useRef(null);
+
+  const registerSliderControl = React.useCallback((element: HTMLElement | null) => {
+    if (element) {
+      controlRef.current = element;
+    }
+  }, []);
+
+  const [valueState, setValueState] = useControlled({
+    controlled: valueProp,
+    default: defaultValue ?? min,
+    name: 'Slider',
+  });
+
+  const { contextValue: compoundComponentContextValue, subitems } = useCompoundParent<
+    string,
+    SliderThumbMetadata
+  >();
+
+  const handleValueChange = React.useCallback(
+    (value: number | number[], thumbIndex: number, event: Event | React.SyntheticEvent) => {
+      if (!onValueChange) {
+        return;
+      }
+
+      // Redefine target to allow name and value to be read.
+      // This allows seamless integration with the most popular form libraries.
+      // https://github.com/mui/material-ui/issues/13485#issuecomment-676048492
+      // Clone the event to not override `target` of the original event.
+      const nativeEvent = (event as React.SyntheticEvent).nativeEvent || event;
+      // @ts-ignore The nativeEvent is function, not object
+      const clonedEvent = new nativeEvent.constructor(nativeEvent.type, nativeEvent);
+
+      Object.defineProperty(clonedEvent, 'target', {
+        writable: true,
+        value: { value, name },
+      });
+
+      onValueChange(value, thumbIndex, clonedEvent);
+    },
+    [name, onValueChange],
+  );
+
+  const range = Array.isArray(valueState);
+
+  const values = React.useMemo(() => {
+    return (range ? valueState.slice().sort(asc) : [valueState]).map((val) =>
+      val == null ? min : clamp(val, min, max),
+    );
+  }, [max, min, range, valueState]);
+
+  const sliderRef = React.useRef<HTMLDivElement>(null);
+
+  const handleRootRef = useForkRef(rootRef, sliderRef);
+
+  const areValuesEqual = React.useCallback(
```

Comment:
@michaldudak The 2nd parameter (the "old" value to be compared against) would be the internal `valueState` which is internal to `useSliderRoot`, I opted to pass this function through the hook instead of passing `valueState` as it seemed messier to pass 3 "values" through the hook - `values`, `percentageValues`, and `valueState`